src/Compilers/CSharp/Test/Emit/Attributes/AttributeTests.cs

```diff
@@ -9202,6 +9493,356 @@ public UnmanagedCallersOnlyAttribute()
+                // (3,21): error CS8773: Feature 'generic attributes' is not available in C# 9.0. Please use language version 10.0 or greater.
+                // public class A<T> : System.Attribute {}
+                Diagnostic(ErrorCode.ERR_FeatureNotAvailableInVersion9, "System.Attribute").WithArguments("generic attributes", "10.0").WithLocation(3, 21),
+                // (5,2): error CS1614: 'A<>' is ambiguous between 'A<T>' and 'AAttribute<T>'; either use '@A<>' or explicitly include the 'Attribute' suffix
+                // [A<int>]
+                Diagnostic(ErrorCode.ERR_AmbiguousAttribute, "A<int>").WithArguments("A<>", "A<T>", "AAttribute<T>").WithLocation(5, 2),
+                // (5,2): error CS8773: Feature 'generic attributes' is not available in C# 9.0. Please use language version 10.0 or greater.
+                // [A<int>]
+                Diagnostic(ErrorCode.ERR_FeatureNotAvailableInVersion9, "A<int>").WithArguments("generic attributes", "10.0").WithLocation(5, 2)
+                );
+        }
+
+        [Fact]
+        public void InheritGenericAttribute()
+        {
+            var source1 = @"
+public class C<T> : System.Attribute { }
+public class D : C<int> { }
+";
+
+            var source2 = @"
+[D]
+public class Program { }
+";
+            var comp = CreateCompilation(new[] { source1, source2 }, parseOptions: TestOptions.Regular9);
+            comp.VerifyDiagnostics(
+                // (2,21): error CS8773: Feature 'generic attributes' is not available in C# 9.0. Please use language version 10.0 or greater.
+                // public class C<T> : System.Attribute { }
+                Diagnostic(ErrorCode.ERR_FeatureNotAvailableInVersion9, "System.Attribute").WithArguments("generic attributes", "10.0").WithLocation(2, 21));
+
+            var comp1 = CreateCompilation(source1);
+            var comp2 = CreateCompilation(source2, references: new[] { comp1.ToMetadataReference() }, parseOptions: TestOptions.Regular9);
+            comp2.VerifyDiagnostics();
+
+            comp2 = CreateCompilation(source2, references: new[] { comp1.EmitToImageReference() }, parseOptions: TestOptions.Regular9);
+            comp2.VerifyDiagnostics();
+        }
+
+        [Fact]
+        public void InheritGenericAttributeInMetadata()
+        {
+            var il = @"
+.class public auto ansi beforefieldinit C`1<T>
+       extends [mscorlib]System.Attribute
+{
+  .method public hidebysig specialname rtspecialname 
+          instance void  .ctor() cil managed
+  {
+    ldarg.0
+    call       instance void [mscorlib]System.Attribute::.ctor()
+    ret
+  }
+}
+
+.class public auto ansi beforefieldinit D
+        extends class C`1<int32>
+{
+    .method public hidebysig specialname rtspecialname
+            instance void  .ctor() cil managed
+    {
+        ldarg.0
+        call        instance void class C`1<int32>::.ctor()
+        ret
+    }
+}
+";
```

Comment:
Consider emitting a library from C# source from the previous test instead. And perhaps merge the two tests.

Reply:
I like having this as a separate test because it can be pasted into an old version of the compiler to demonstrate that the feature didn't cause any behavior changes in this scenario.